freqtrade/exchange/common.py

```diff
@@ -85,7 +85,7 @@ def calculate_backoff(retrycount, max_retries):
 def retrier_async(f):
     async def wrapper(*args, **kwargs):
         count = kwargs.pop('count', API_RETRY_COUNT)
-        kucoin = args[0].name == "Kucoin"  # Check if the exchange is KuCoin.
+        kucoin = args[0].name.lower() == "kucoin"  # Check if the exchange is KuCoin.
```

Comment:
Please align it to how it's done in the async retrier now (using proper casing).

Also note that the test is still outstanding as requested below.